plugins/iamauth.js

```diff
@@ -0,0 +1,222 @@
+// Copyright © 2017 IBM Corp. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+'use strict';
+
+const async = require('async');
+const debug = require('debug')('cloudant:plugins:iamauth');
+const lockFile = require('lockfile');
+const request = require('request');
+const tmp = require('tmp');
+const u = require('url');
+
+const BasePlugin = require('./base.js');
+
+function noop() {}
```

Comment:
Is this really needed? In the single usage of it in `withLock` can't we just `callback(error)`?

Reply:
Added in 4a03a1a.